Bug! Uploading recordings with UTF-8 Chars in the title fails.

This is quite serious.
We've just noticed that the upload/ingest for recordings that have UTF-8 chars in the title (Umlaute, etc.) fails reproducable. This is tested against Opencast 2.2.3.

```
Apr 18 16:48:12 ele-ca-02 pyca[10980]: [pyca.utils:181:recording_state()] b'8385795844570640384 set to uploading'
Apr 18 16:48:12 ele-ca-02 pyca[10980]: [pyca.ingest:95:ingest()] Selecting ingest service to use: http://electuresda1.uni-muenster.de:8080/ingest
Apr 18 16:48:12 ele-ca-02 pyca[10980]: [pyca.ingest:98:ingest()] Creating new mediapackage
Apr 18 16:48:12 ele-ca-02 pyca[10980]: [pyca.ingest:103:ingest()] Adding episode DC catalog
Apr 18 16:48:12 ele-ca-02 pyca[10980]: [pyca.ingest:69:start_ingest()] Something went wrong during the upload
Apr 18 16:48:12 ele-ca-02 pyca[10980]: [pyca.ingest:70:start_ingest()] Traceback (most recent call last):
Apr 18 16:48:12 ele-ca-02 pyca[10980]:   File "/usr/lib/python3.6/site-packages/pyca/ingest.py", line 67, in start_ingest
Apr 18 16:48:12 ele-ca-02 pyca[10980]:     workflow_config)
Apr 18 16:48:12 ele-ca-02 pyca[10980]:   File "/usr/lib/python3.6/site-packages/pyca/ingest.py", line 110, in ingest
Apr 18 16:48:12 ele-ca-02 pyca[10980]:     mediapackage = http_request(service + '/addDCCatalog', fields)
Apr 18 16:48:12 ele-ca-02 pyca[10980]:   File "/usr/lib/python3.6/site-packages/pyca/utils.py", line 52, in http_request
Apr 18 16:48:12 ele-ca-02 pyca[10980]:     curl.setopt(curl.HTTPPOST, post_data)
Apr 18 16:48:12 ele-ca-02 pyca[10980]: pycurl.error: (0, '')
Apr 18 16:48:12 ele-ca-02 pyca[10980]: [pyca.utils:181:recording_state()] b'8385795844570640384 set to upload_error'
```
Recording cannot be ingested at all, tried forcing the status back to `FINISHED_RECORDING` to trigger a new ingest.

Logs on the Opencast server do not show anything suspicious:
```
...
Apr 18 18:48:12 xxx docker[10270]: 2017-04-18 18:48:12,937 | INFO  | (IngestServiceImpl:648) - Created mediapackage 4fd6cc75-8103-4cff-a21c-b5d7105fdf15
...
```

Ingesting recordings without any special chars works just fine.

I still get an error when the recording is finished

Apr 26 13:53:03 gr-hs-pi [pyca.utils:174:recording_state()] [INFO] 8088895241510331392 set to uploading
Apr 26 13:53:03 gr-hs-pi [pyca.ingest:82:ingest()] [INFO] Selecting ingest service to use: http://oc-admin.rrze.uni-erlangen.de:8080/ingest
Apr 26 13:53:03 gr-hs-pi [pyca.ingest:85:ingest()] [INFO] Creating new mediapackage
Apr 26 13:53:03 gr-hs-pi [pyca.ingest:99:ingest()] [INFO] Adding episode DC catalog
Apr 26 13:53:03 gr-hs-pi [pyca.ingest:57:start_ingest()] [ERROR] Something went wrong during the upload
Apr 26 13:53:03 gr-hs-pi [pyca.ingest:58:start_ingest()] [ERROR] Traceback (most recent call last):#012  File "pyca/ingest.py", line 55, in start_ingest#012    ingest(event.get_tracks(), event.directory(), event.uid, attachments)#012  File "pyca/ingest.py", line 103, in ingest#012    mediapackage = http_request(service + '/addDCCatalog', fields)#012  File "pyca/utils.py", line 52, in http_request#012    curl.setopt(curl.HTTPPOST, post_data)#012error: (0, '')

--- a/pyca/ingest.py
+++ b/pyca/ingest.py
@@ -71,7 +71,7 @@ def ingest(event):
             logger.info('Adding %s DC catalog' % name)
             fields = [('mediaPackage', mediapackage),
                       ('flavor', 'dublincore/%s' % name),
-                      ('dublinCore', data)]
+                      ('dublinCore', data.encode('utf-8'))]
             mediapackage = http_request(service + '/addDCCatalog', fields)
 
     # add track

